src/toil/jobStores/utils.py

```diff
@@ -109,3 +110,33 @@ def single_attempt():
             yield
 
         yield single_attempt()
+
+default_delays = (0, 1, 1, 4, 16, 64)
+default_timeout = 300
+
+
+def retry_http(delays=default_delays, timeout=default_timeout, predicate=None):
+    """
+    >>> i = 0
+    >>> for attempt in retry_http():
+    ...     with attempt:
+    ...         i += 1
+    ...         raise urllib2.HTTPError('http://www.test.com', '408', 'some message', {}, None)
+    Traceback (most recent call last):
+    ...
+    HTTPError: HTTP Error 408: some message
+    >>> print i
+    3
+    >>> for attempt in retry_http():
+    ...     with attempt:
+    ...         i += 1
+    ...         raise urllib2.HTTPError('http://www.test.com', '503', 'some message', {}, None)
+    Traceback (most recent call last):
+    ...
+    HTTPError: HTTP Error 503: some message
+    >>> print i
+    6
+    """
+    def retryable_http_error(e):
+        return isinstance(e, urllib2.HTTPError) and e.code in ('503', '408')
```

Comment:
Should I add the 500 case to the doctest? I think its pretty obvious what will happen, and it might just be too much duplicated code to add. Also should there be any documentation with links to what these response codes mean?